### PR TITLE
SINGA -475 add Log operator to singa

### PR DESCRIPTION
**Pull request by pinpom**

### CODE REVIEW

File: test/python/test_operation.py [hunk 1]
```diff
@@ -610,6 +610,18 @@ def test_Atanh_gpu(self):
...
+    def test_Log(self):
+        X=np.array([1, np.e, np.e**2, np.e**4]).reshape(2,2).astype(np.float32)
+        XT=np.log(X)
+        x=tensor.from_numpy(X)
+        x.to_device(gpu_dev)
+
+        result=autograd.log(x)
+        dx=result.creator.backward(x.data)
+
+        np.testing.assert_array_almost_equal(tensor.to_numpy(result), XT)
```
Comment (joddiy) [inline test/python/test_operation.py]: the backward of log hasn't been tested.

Comment (pinpom) [reply test/python/test_operation.py]: added backward test.

**Comment by joddiy** [comment]
ready for merge

**Comment by chrishkchris** [comment]
I have run the test in docker container using cloud K80. It is ready for merge.

**Comment by nudles** [comment]
there are conflicts..
two solutions:
1. [merge this branch with the latest master](https://docs.fast.ai/dev/git.html#how-to-keep-your-feature-branch-up-to-date).
2. checkout a new branch from the [latest master](https://github.com/apache/incubator-singa/) ; copy the added code into the autograd.py and test_operations.py; send a new PR.

**Comment by chrishkchris** [comment]
Yes, could pinpom please merge your branch with master branch, or send a new pr after copying the new codes.

I guess this maybe because different PRs insert the new codes at the same line, so git detected it as different ways of changing code and hence conflicts.

**Pull request merged**